jass/lib/PlayerAlliance.j

```diff
@@ -0,0 +1,197 @@
+                    //add vision
+                    call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_VISION, true)
+                    call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_VISION, true)
+                    //check for shared control
+                    if (flag >= ALLIANCE_ALLIED_UNITS) then
+                        //add shared control
+                        call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_CONTROL, true)
+                        call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_CONTROL, true)
+                        //check for advanced control
+                        if (flag == ALLIANCE_ALLIED_ADVUNITS) then
+                            call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_ADVANCED_CONTROL, true)
+                            call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_ADVANCED_CONTROL, true)
+                        else
+                            call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_ADVANCED_CONTROL, false)
+                            call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_ADVANCED_CONTROL, false)
+                        endif
+                    else
+                        call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_CONTROL, false)
+                        call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_CONTROL, false)
+                        call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_ADVANCED_CONTROL, false)
+                        call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_ADVANCED_CONTROL, false)
+                    endif
+                else
+                    call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_VISION, false)
+                    call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_VISION, false)
+                    call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_CONTROL, false)
+                    call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_CONTROL, false)
+                    call SetPlayerAlliance(p1, p2, ALLIANCE_SHARED_ADVANCED_CONTROL, false)
+                    call SetPlayerAlliance(p2, p1, ALLIANCE_SHARED_ADVANCED_CONTROL, false)
+                endif
+            endif
+        debug else
+            debug if (p1 == null) then
+                debug call DisplayTimedTextToPlayer(GetLocalPlayer(), 0, 0, 60, "PLAYER ALLIANCE ERROR: NULL PLAYER 1 ")
+            debug endif
+            debug if (p2 == null) then
+                debug call DisplayTimedTextToPlayer(GetLocalPlayer(), 0, 0, 60, "PLAYER ALLIANCE ERROR: NULL PLAYER 2 ")
+            debug endif
+            debug if (p1 == p2) then
+                debug call DisplayTimedTextToPlayer(GetLocalPlayer(), 0, 0, 60, "PLAYER ALLIANCE ERROR: PLAYER 1 == PLAYER 2 ")
+            debug endif
+            debug if (flag < 0 or flag > 7) then
+                debug call DisplayTimedTextToPlayer(GetLocalPlayer(), 0, 0, 60, "PLAYER ALLIANCE ERROR: INVALID ALLIANCE FLAG")
+            debug endif
+        debug endif
+    endfunction
+endlibrary
```

Comment:
Do we really need this library?

I can see us using it [here](https://github.com/island-troll-tribes/island-troll-tribes/blob/master/jass/systems/class-selection.zn#L256) but I'm not sure what the advantage would be.